### PR TITLE
Rework async content fetching

### PR DESCRIPTION
**Pull request by bh2smith**
In order to make the diff in https://github.com/cowprotocol/dune-sync/pull/44 easier to digest, we do a small refactoring before. 

We break async content fetching into two methods.

### CODE REVIEW

File: src/fetch/orderbook.py [hunk 1]
```diff
@@ -49,9 +49,6 @@ def _pg_engine(db_env: OrderbookEnv) -> Engine:
     def _read_query_for_env(
         cls, query: str, env: OrderbookEnv, data_types: Optional[dict[str, str]] = None
     ) -> DataFrame:
-        # It seems there is a bug in mypy on the dtype field (with error [arg-type]).
-        # They expect types that should align with what we pass.
-        # More context at: https://github.com/cowprotocol/dune-sync/issues/24
         return pd.read_sql_query(query, con=cls._pg_engine(env), dtype=data_types)
```
Comment (bh2smith) [inline src/fetch/orderbook.py:52]: This was supposed to be removed in #46 but forgotten. Slipping it in here. Also related to #24 (which is now closed)

File: src/fetch/ipfs.py [hunk 2]
```diff
@@ -76,50 +77,64 @@ async def fetch_many(  # pylint: disable=too-many-locals
...
+        if previous_attempts:
+            log.debug(f"still {base_message} {total_attempts} attempts")
+        else:
+            log.debug(f"{base_message} {max_retries} retries")
+
+        return NotFoundContent(
+            app_hash=self.hex,
+            first_seen_block=first_seen_block,
+            attempts=total_attempts,
+        )
```
[the remark this answers is not in the document]
Comment (bh2smith) [reply src/fetch/ipfs.py:140]: I will need to check it out, but at this particular moment, I am not sure how to model what we have as a decorator. 

In this scenario we are trying to don things asyncronously with a special library and we are working with data loaded from a previous run of the program. In my opinion there is quite a bit of non-generic logic here that I suspect may fit so nicely in someone else's framework.

It does appear what you linked here supports async which is cool... Would be happy to review a PR if you wanted to take a stab at doing this though.

[the remark this answers is not in the document]
Comment (gentrexha) [reply src/fetch/ipfs.py:140]: > It does appear what you linked here supports async which is cool... Would be happy to review a PR if you wanted to take a stab at doing this though.

Totally. I'll create a PR for it!

[the remark this answers is not in the document]
Comment (bh2smith) [reply src/fetch/ipfs.py:140]: I'll transform your initial comment into an issue so you can "close it" with your PR:

https://github.com/cowprotocol/dune-sync/issues/48

**Pull request merged**